### PR TITLE
Make (STATUS DATE) return a four-digit year, and fix users of it.

### LINKED ISSUE #1096: Y2K problems in TIME breaking Macsyma build
Hello,

I did in the morning a git pull, and I got a build timeout using KLH10:

```
MCLDAT 
;GC DUE TO USER
; 4169.[15%] LIST, 3337.[72%] FIXNUM, 494.[96%] FLONUM, 
; 507.[99%] BIGNUM, 312.[4%] SYMBOL, 424.[82%] HUNK8, 
; 432.[84%] HUNK16, 434.[84%] ARRAY WORDS FREE
;GC-DAEMON: cons-rate%[oldgcsize->marked/gcsize/spcsize]
; Consed=5743, Marked=31513, Allocated/Marked=1.0, Memfree= 26624
; LIST 74%[30208->23057/33280/27136]  FIXNUM 24%[4608->1331/5120/4608]  FLONUM 0%[512->18/512]  BIGNUM 0%[512->5/512]  
; SYMBOL 0%[7680->6856/7168]  HUNK8 0%[512->88/512]  HUNK16 0%[512->80/512]  ARRAY 0%[512->78/512]  

(SETQ ERRSET (FUNCTION ERRSET-BREAK)) => ERRSET-BREAK 
(STATUS MEMFREE) => 26624. 
(STATUS FEATURES) => (IOTA GRINDEF DEFSTRUCT FORMAT FASLAP NCOMPLR COMPLR MACLISP PDP10 BIGNUM FASLOAD HUNK FUNARG ROMAN SFA PAGING!
 NEWIO DB ITS) 
MAXMAC 227
DEFMAX 98
EVONCE 14
DEBUG 69
DEFINE 65
EXTHUK 34
MAILER 2
FORMAT 827.
EXPAND 10
MOPERS 49
MCOMPILER 32.
MFORMA 104
LMMAC 87
MLMAC 92
EXTSTR 96
MACAID 120
LET 98
BREAKLEVEL 
ERRCK 30
NUMERM 21
TIME 4
DEFSTRUCT 280
GRINDEF 462
MACSYMA-MODULE 9
SMURF 
BLTARRAY 2
PROCS 16
DEFOPT 8
UMLMAC 42
QUERIO 51
BACKQ 53
DEFMACRO 166
USER-QUERY 6
COMPLR 2155
DEFVSY 84
SETF 293
TRANSM 129
MLSUB 17
MCL 59
DCL 7/26/18 12:42:01
IOTA 40
GETMIDASOP 17
ERMSGC 210
SHARABLE 47
SHARPM 82
YESNOP 44
;Loading CERROR 47
Today is a Friday, not a Thursday.
;NIL ?

;BKPT *RSET-TRAP

The last command timed out.
Makefile:49: recipe for target 'out/klh10/rp0.dsk' failed
make: *** [out/klh10/rp0.dsk] Error 1
```

What does it mean "Today is a Friday, not a Thursday." ? :-)

regards,
Fausto

### PR DESCRIPTION
**Pull request by atsampson**
Fixes #1096. A full build with simh now completes.

I've removed duplicate versions of three files which didn't seem to be referenced anywhere, and when bumping a version number such as DNA42, I've gone to 43 rather than DNA43 (since it's now collectively maintained).

### CODE REVIEW

File: build/build.tcl [hunk 1]
```diff
@@ -72,17 +72,22 @@ proc ip_address {string} {
     format "%o" $x
 }
 
+# Respond to the output from (load ...).
+proc respond_load { r } {
+    expect -re {[\r\n][0-9]+\.? *[\r\n]}
+    type $r
+}
+
```
Comment (eswenson1) [inline build/build.tcl]: That is *such* a good idea.  You don't know how many times that size/offset has changed and broken my build (attempts).

File: src/libdoc/phsprt.gls17 [hunk 2]
```diff
@@ -1,235 +0,0 @@
-
```
Comment (eswenson1) [inline src/libdoc/phsprt.gls17]: Why is this file deleted?

Comment (eswenson1) [reply src/libdoc/phsprt.gls17]: Otherwise, this looks great and I approve it.

Comment (atsampson) [reply src/libdoc/phsprt.gls17]: PHSPRT GLS17 and PHSPRT GLS18 were both in the repository - GLS17 doesn't seem to be used explicitly, and GLS18 is updated in a later commit. The difference between the two is as follows:

```
--- src/libdoc/phsprt.gls17	2018-07-27 22:51:03.968255753 +0100
+++ src/libdoc/phsprt.gls18	2018-07-27 22:51:03.968255753 +0100
@@ -24,7 +24,9 @@
 ;;; It takes zero to two arguments, one being the sun position
 ;;; (the output of SUN in BKPH's SUN package), and the other a file.
 
-(declare (*lexpr timestamp datimprinc phaseprinc sunposprinc) (newio t))
+(declare (*lexpr timestamp datimprinc phaseprinc sunposprinc)
+ ;;;(newio t)
+)
 
 (defprop moonphase (phase fasl dsk liblsp) autoload)
 (defprop sun-now-here (sun fasl dsk liblsp) autoload)
```

Comment (eswenson1) [reply src/libdoc/phsprt.gls17]: Ah, right.  I had kept the old version there temporarily, but forgot to delete it. Thanks.

File: build/build.tcl (hunk 1 above)
Comment (atsampson) [reply build/build.tcl]: It took several tries to come up with a Tcl regexp that works - I ended up writing an expect script to test it against ITS over telnet...

**Comment by larsbrinkhoff** [comment]
Did you consider RQDATE rather than .RDATE and .RYEAR?  No need to change anything, just curious.

**Comment by atsampson** [comment]
I did consider RQDATE but couldn't see a strong argument either way - I think SDATE would be about the same length written with either. Maybe revisit this when we sort out the 2027 problem...

**Comment by eswenson1** [comment]
Thanks! You want to submit that MCR? Or at least enter a ticket (http://multics-trac.swenson.org/newticket)?

**Comment by larsbrinkhoff** [comment]
I submitted a change request to Kent Pitman about updating maclisp.info.  I think it would lend weight to my request if Multics Maclisp was also fixed.

**Comment by eswenson1** [comment]
Yeah. I’m happy to fix the Multics stuff. I’m working on the MR12.6g release now.

**Comment by eswenson1** [comment]
And I allocated an MCR number (MCR10051) for the purpose of this (STATUS DATE) fix on Multics.  When I have some time, I'll write it up (and an MTB if required) and get approval to make changes for MR12.6g.

**Comment by larsbrinkhoff** [comment]
Hello @g000001,

For you information:  This GitHub pull request is a backward incompatible fix for a Y2K problem.  All Lisp code has been updated accordingly.  Multics Maclisp has also been fixed.

**Comment by g000001** [comment]
@larsbrinkhoff thank you for your infomation :smiley:
